packages/vite/src/node/plugins/resolve.ts

```diff
@@ -725,39 +725,42 @@ export function resolvePackageEntry(
         }
       }
     }
+    entryPoint = entryPoint || data.main
+
+    // entry also emtry, should try default entry
+    // https://nodejs.org/api/modules.html#all-together
+    for (let entry of entryPoint
+      ? [entryPoint]
+      : ['index.js', 'index.json', 'index.node']) {
```

Comment:
:eyes: 

Could you split this into separat lines?

```suggestion
    const entryPoints = entryPoint
      ? [entryPoint]
      : ['index.js', 'index.json', 'index.node']
    for (let entry of entryPoints) {
```